chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -436,24 +441,24 @@
     end
 
     # Index known barclamp repositories for this OS
-    node[:provisioner][:repositories][os] ||= Mash.new
-    node[:provisioner][:repositories][os][arch] ||= Mash.new
+    repositories[os] = Mash.new
+    repositories[os][arch] = Mash.new
 
     if File.exist?("#{os_dir}/crowbar-extra") && File.directory?("#{os_dir}/crowbar-extra")
       Dir.foreach("#{os_dir}/crowbar-extra") do |f|
         next unless File.symlink? "#{os_dir}/crowbar-extra/#{f}"
-        node[:provisioner][:repositories][os][arch][f] ||= Hash.new
+        repositories[os][arch][f] = Hash.new
```

Comment:
You are using `=` instead of `||=` - couldn't that be a problem, given this is a loop?